assert/assertions_test.go

```diff
@@ -182,49 +177,37 @@ func TestIsType(t *testing.T) {
 
 }
 
-type myType string
-
 func TestEqual(t *testing.T) {
+	type myType string
 
 	mockT := new(testing.T)
-
-	if !Equal(mockT, "Hello World", "Hello World") {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, 123, 123) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, 123.5, 123.5) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, []byte("Hello World"), []byte("Hello World")) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, nil, nil) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, int32(123), int32(123)) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, uint64(123), uint64(123)) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, myType("1"), myType("1")) {
-		t.Error("Equal should return true")
-	}
-	if !Equal(mockT, &struct{}{}, &struct{}{}) {
-		t.Error("Equal should return true (pointer equality is based on equality of underlying value)")
-	}
 	var m map[string]interface{}
-	if Equal(mockT, m["bar"], "something") {
-		t.Error("Equal should return false")
-	}
-	if Equal(mockT, myType("1"), myType("2")) {
-		t.Error("Equal should return false")
+
+	cases := []TestCase{
+		{"Hello World", "Hello World", true},
+		{123, 123, true},
+		{123.5, 123.5, true},
+		{[]byte("Hello World"), []byte("Hello World"), true},
+		{nil, nil, true},
+		{int32(123), int32(123), true},
+		{uint64(123), uint64(123), true},
+		{myType("1"), myType("1"), true},
+		{&struct{}{}, &struct{}{}, true}, // pointer equality is based on equality of underlying value
```

Comment:
This is one case where there was a specific message about the test. Not sure what to do with that (though I do add testcase specific messages further down below)

Reply:
I think we can handle that if we define the `TestCase` struct on a per-test level rather than package level, as mentioned in the comment above.